File: algorithms/1704.determine-if-string-halves-are-alike/solution.rb
VOWELS = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']

# @param {String} s
# @return {Boolean}
def halves_are_alike(s)
  a_vowels, b_vowels = 0, 0

  mid = s.size / 2
  (0...mid).each do |i|
    a_vowels += 1 if VOWELS.include?(s[i])
    b_vowels += 1 if VOWELS.include?(s[-i-1])
  end

  a_vowels == b_vowels
end

File: algorithms/1704.determine-if-string-halves-are-alike/solution.cpp
#include <string>
#include <unordered_set>

using namespace std;

class Solution {
public:
    bool halvesAreAlike(string s) {
        unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};

        int length = s.length();
        int mid = length / 2;
        int aVowels = 0;
        int bVowels = 0;
        for (int i = 0; i < length; i++) {
            if (vowels.count(s[i])) {
                if (i < mid) {
                    aVowels++;
                } else {
                    bVowels++;
                }
            }
        }

        return aVowels == bVowels;
    }
};
